### PR TITLE
Shared flush

### PR DESCRIPTION
**Pull request by mcfadden8**
Work to address #489

### CODE REVIEW

File: src/scr_util.c [hunk 1]
```diff
@@ -389,3 +390,23 @@ spath* scr_get_prefix(const char* str)
 
   return prefix_path;
 }
+
+/* TODO: This needs to be moved somewhere else and probabbly renamed as well
+ */
+int scr_leader_rank(scr_filemap* map, const char *file)
```
Comment (adammoody) [inline src/scr_util.c]: I think we could move this to scr_filemap.c.

File: src/scr_util.c [hunk 2]
```diff
@@ -389,3 +390,23 @@ spath* scr_get_prefix(const char* str)
...
+    int group_ranks;
+    int group_rank;
+
+    scr_filemap_get_meta(map, file, meta);
+    scr_meta_get_group_ranks(meta, &group_ranks);
+    scr_meta_get_group_rank(meta, &group_rank);
+
+    scr_meta_delete(&meta);
+
+    int rc = group_ranks == 1 || group_rank == 0;
```
Comment (adammoody) [inline src/scr_util.c]: We can simplify this to ``int rc = group_rank == 0;`` and then it's not necessary to get the number of ranks.

Comment (mcfadden8) [reply src/scr_util.c]: Done

File: src/scr_util.c (hunk 1 above)
Comment (mcfadden8) [reply src/scr_util.c]: Done

**Comment by mcfadden8** [comment]
This PR is being closed/replaced by PR #511

**Pull request closed without merging**